subsys/bluetooth/host/att.c

```diff
@@ -3249,15 +3190,6 @@ static void bt_att_released(struct bt_l2cap_chan *ch)
 {
 	struct bt_att_chan *chan = ATT_CHAN(ch);
 
-	/* Traverse the ATT bearer's TX queue and free the metadata. */
-	while (!sys_slist_is_empty(&chan->tx_cb_queue)) {
-		sys_snode_t *tx_meta_data_node = sys_slist_get(&chan->tx_cb_queue);
-		struct bt_att_tx_meta_data *tx_meta_data = CONTAINER_OF(
-			tx_meta_data_node, struct bt_att_tx_meta_data, tx_cb_queue_node);
-
-		tx_meta_data_free(tx_meta_data);
-	}
-
```

Comment:
We don't need to free any metadata here anymore?

Reply:
Yes. The metadata storage is no longer dynamic. There is now a metadata object statically associated with each net_buf object in the pool. It's as if the metadata is part of the net_buf now.

Reply:
It was never freed here. But yes, that should be the case.